### PR TITLE
Add Autoscaler to CSE

### PR DESCRIPTION
**Pull request by adambarreiro**
## Overview
This PR automates the steps to deploy an Autoscaler into a Kubernetes cluster stated in this document:
https://www.vmware.com/content/dam/digitalmarketing/vmware/en/pdf/docs/vmw-whitepaper-cluster-auto-scaler.pdf

This procedure affects cluster creation and cluster update operations.

All the logic is implemented in https://github.com/vmware/go-vcloud-director/pull/678

## Description

The Go SDK implementation now adds extra fields to work with the Autoscaler in the worker pools.
Hence, the `worker_pool` block has two new fields, `autoscaler_max_replicas` and `autoscaler_min_replicas`, that when set, they deploy the Autoscaler to the cluster to start the automatic adjustment of its nodes.

When `autoscaler_max_replicas` and `autoscaler_min_replicas` are set, `machine_count` must be either unset or `0`, and viceversa: when `machine_count` is set, `autoscaler_max_replicas` and `autoscaler_min_replicas` must be `0`.

As it can be seen in the [Go SDK implementation](https://github.com/vmware/go-vcloud-director/pull/678), once the autoscaler is deployed to the cluster, it is not removed. It's just scaled to 0 or 1 (depending on the mentioned arguments).

## Tests
Updated `TestAccVcdCseKubernetesCluster` (the modifications check Autoscaler updates) and created`TestAccVcdCseKubernetesClusterCreationWithAutoscaler` (this one checks cluster creation with Autoscaler). Both check that the autoscaler is created and the information is retrieved (set in the structure) correctly.

```
TEST_VCD_CSE=1 go test -tags cse -run '^Test_Cse' -v -timeout 0
```

Anyway, tests only check that the YAML definition is correct. One should check that the autoscaler works, manually:

```
kubectl --kubeconfig=kubeconfig-n kube-system describe deploy cluster-autoscaler
kubectl --kubeconfig=kubeconfig-n kube-system get po | grep autoscaler
kubectl --kubeconfig=kubeconfig -n kube-system logs cluster-autoscaler-xxxxxxxx-xxxx -f
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -70,3 +70,5 @@ require (
 	google.golang.org/grpc v1.63.2 // indirect
```
Comment (Didainius) [inline go.mod:70]: Missing changelog record

**Review by Didainius — COMMENTED** [review]
Running test `TestAccVcdCseKubernetesClusterCreationWithAutoscaler`

**Review by Didainius — APPROVED** [review]
Test passed. The only question I have is:

Removal process will remove the CSE cluster that might come out with default envs.


```
Start leftovers removal
[vcd_nsxt_alb_controller] cse_alb_controller (keep)
[vcd_provider_vdc] pvdc-2024-06-05-13-36-12.485 (keep)
[vcd_provider_vdc] nsxTPvdc1 (keep)
[vcd_network_pool] NSX-T Overlay 1 (keep)
[vcd_network_pool] unicastNetworkPoolName (keep)
[vcd_org] System (keep)
[vcd_org] tenant_org (keep)
        [vcd_org_vdc] tenant_vdc (keep)
                [vcd_vapp] test1 (DELETE)
                 REMOVING vApp tenant_vdc/test1
[vcd_org] solutions_org (keep)
        [vcd_org_vdc] solutions_vdc (keep)
                [vcd_rde] test1 (DELETE)
                 REMOVING RDE test1 WITH TYPE urn:vcloud:type:vmware:capvcdCluster:1.3.0
ok      github.com/vmware/terraform-provider-vcd/v3/vcd 3241.561s
```

File: go.mod (hunk 1 above)
Comment (adambarreiro) [reply go.mod:70]: Oops, thanks

File: go.mod [hunk 2]
```diff
@@ -70,3 +70,5 @@ require (
...
 	google.golang.org/protobuf v1.34.0 // indirect
 )
+
+replace github.com/vmware/go-vcloud-director/v2 => github.com/adambarreiro/go-vcloud-director/v2 v2.17.0-alpha.1.0.20240606095202-a2a846adaa90
```
Comment (dataclouder) [inline go.mod:74]: reminder to remove this replacement before merging

**Review by dataclouder — APPROVED** [review]

File: website/docs/r/cse_kubernetes_cluster.html.markdown [hunk 3]
```diff
@@ -174,12 +174,78 @@ Each block asks for the following arguments:
...
+* `autoscaler_max_replicas` - (Optional; *v3.13+*) Together with `autoscaler_min_replicas`, and **only when `machine_count=0` (or unset)**, defines the maximum number of nodes that
+  the Kubernetes Autoscaler will deploy for this worker pool. Read the section below for details.
+* `autoscaler_min_replicas` - (Optional; *v3.13+*) Together with `autoscaler_max_replicas`, and **only when `machine_count=0` (or unset)**, defines the minimum number of nodes that
+  the Kubernetes Autoscaler will deploy for this worker pool. Read the section below for details.
+
+#### Worker pools with Kubernetes Autoscaler enabled
+
+-> Supported in provider *v3.13+*
+
+~> The Autoscaler can only work in clusters with Internet access, as it needs to download the Docker image from
```
Comment (lvirbalas) [inline website/docs/r/cse_kubernetes_cluster.html.markdown:193]: If there's no access to Internet, is it possible to provide the image manually?

**Review by lvirbalas — APPROVED** [review]

File: go.mod (hunk 2 above)
Comment (adambarreiro) [reply go.mod]: Done

**Pull request merged**